src/onewire.rs

```diff
@@ -0,0 +1,102 @@
+use core::marker::PhantomData;
+use core::ptr;
+
+use esp_idf_sys::*;
+
+use crate::peripheral::Peripheral;
+
+#[derive(Debug)]
+pub struct Device<'b> {
+    _device: onewire::onewire_device_t,
+    // _bus: &'b BusDriver<'b>, // not sure how I would hold a reference to this aside from the handle
+    _p: PhantomData<&'b ()>, // holds the lifetime since the device is linked to the lifetime of the bus
+}
+
+impl<'a> Device<'a> {
+    fn new(device_handle: onewire::onewire_device_t) -> Self {
+        Self {
+            _device: device_handle,
+            _p: PhantomData,
+        }
+    }
+
+    /// get the handle of the bus the device is attached to
+    pub fn bus(&self) -> onewire::onewire_bus_handle_t {
+        self._device.bus
+    }
+
+    /// get the device address
+    pub fn address(&self) -> u64 {
+        self._device.address
+    }
+}
+
+/// wrapper around the device iterator to search for available devices on the bus
+pub struct DeviceSearch {
```

Comment:
In your example you don't use the Device nor the DeviceSearch. If you create helper structs for you that you in the end only use internally yourself don't make them pub structs.

Reply:
I do use the public methods of these structs (eg `next_device()` and `address()`), so I think they should be exposed.

Reply:
I have annotated the types a bit clearer in the example to show where the public interface is